### PR TITLE
Make Draggable use gestures

### PR DESCRIPTION
**Pull request by Hixie**
Draggable is now itself a gesture arena member. This means it won't
conflict with other gesture recognisers in the same path.

This also allows variants of Draggable that are triggered by other
gestures.

Also, some cleanup of DoubleTapGestureRecognizer, GestureDetector, and
PrimaryPointerGestureRecognizer.

Also, make MultiTapGestureRecognizer support a timeout for longpress.

Also, make Draggable data be typed.

Also, hide warnings about constructor warnings for now. Analyzer doesn't
support them yet. (Have to do this on a per-line basis)

Directions for future research:
 - animating the avatar (enter/exit transitions)
 - interaction with the navigator (canceling a drag on page navigation, etc)
 - double-tap draggable

### CODE REVIEW

File: sky/tools/skyanalyzer [hunk 1]
```diff
@@ -36,14 +36,17 @@ _IGNORED_PATTERNS = [
...
   re.compile(r'^\[error\] Type check failed: .*\(dynamic\) is not of type'), # allow unchecked casts from dynamic
 
-  # String mode hints. Those are going away anyway, but these in particular aren't useful.
-  re.compile(r'^\[hint\] .+ requires dynamic invoke'), # too many false positives, e.g. https://github.com/dart-lang/sdk/issues/24564
-  re.compile(r'^\[hint\] Runtime check on non-ground type .+ may throw StrongModeError'), # https://github.com/dart-lang/sdk/issues/24565
-  re.compile(r'^\[hint] \([_, ]+\) .+ has inferred type '), # ignore underscore-only arguments
+  # # String mode hints. Those are going away anyway, but these in particular aren't useful.
+  # re.compile(r'^\[hint\] .+ requires dynamic invoke'), # too many false positives, e.g. https://github.com/dart-lang/sdk/issues/24564
+  # re.compile(r'^\[hint\] Runtime check on non-ground type .+ may throw StrongModeError'), # https://github.com/dart-lang/sdk/issues/24565
+  # re.compile(r'^\[hint] \([_, ]+\) .+ has inferred type '), # ignore underscore-only arguments
```
Comment (krisgiesing) [inline sky/tools/skyanalyzer]: Are these changes related to the rest of the commit?

Comment (Hixie) [inline sky/tools/skyanalyzer]: No, those are just cleaning up old stuff. Those hints aren't in the analyzer anymore.

Comment (Hixie) [inline sky/tools/skyanalyzer]: I'll delete them in a separate patch.

File: sky/packages/sky/lib/src/gestures/arena.dart [hunk 2]
```diff
@@ -129,20 +137,32 @@ class GestureArena {
...
+      if (!state.isOpen)
+        _tryToResolveArena(key, state);
     } else {
       assert(disposition == GestureDisposition.accepted);
-      _arenas.remove(key);
-      for (GestureArenaMember rejectedMember in state.members) {
-        if (rejectedMember != member)
-          rejectedMember.rejectGesture(key);
+      if (state.isOpen) {
+        state.eagerWinner = member;
```
Comment (abarth) [inline sky/packages/sky/lib/src/gestures/arena.dart]: What if there's already an eagerWinner?  Usually we have the first one in the event path win, which would mean whoever gets into the eagerWinner slot first.

Comment (Hixie) [inline sky/packages/sky/lib/src/gestures/arena.dart]: I intended to make the first one win, this was an oversight. Will fix.

**Comment by abarth** [comment]
LGTM

**Comment by krisgiesing** [comment]
Can we add tests for the eager-winner rules?

**Comment by Hixie** [comment]
Sure. What should I test?

**Comment by krisgiesing** [comment]
That the eager winner chosen is the first one, mainly (since that was a point of confusion earlier).  Probably also that the eager winner is chosen at the right point in the sequence.

**Comment by Hixie** [comment]
Done: https://github.com/flutter/engine/pull/1976
